sharing_portal/views.py

```diff
@@ -215,19 +241,49 @@ def edit_artifact(request, artifact):
             version_id = request.POST.get("delete_version")
             try:
                 version = artifact.versions.get(pk=version_id)
-                if not version.doi:
-                    version.delete()
-                    messages.add_message(request, messages.SUCCESS,
-                        'Successfully deleted artifact version.')
-                else:
-                    messages.add_message(request, messages.ERROR,
-                        'Cannot delete versions already assigned a DOI.')
+                _delete_artifact_version(request, version)
             except ArtifactVersion.DoesNotExist:
-                messages.add_message(request, messages.ERROR,
-                    'Artifact version {} does not exist'.format(version_id))
+                messages.add_message(
+                    request,
+                    messages.ERROR,
+                    "Artifact version {} does not exist".format(version_id),
+                )
             # Return to edit form
             return HttpResponseRedirect(
-                reverse('sharing_portal:edit', args=[artifact.pk]))
+                reverse("sharing_portal:edit", args=[artifact.pk])
+            )
+
+        elif "delete_all" in request.POST:
+            for version in artifact.versions:
+                if not _delete_artifact_version(request, version):
+                    messages.add_message(
+                        request,
+                        messages.ERROR,
+                        "Could not delete all artifact versions.",
+                    )
+                    return HttpResponseRedirect(
```

Comment:
Should it attempt to delete all versions before erroring? If one version has a doi, but the others don't, this will stop at the first version, but it could continue.

Reply:
I think regardless, we don't want to delete artifacts with undeleted versions, so it shouldn't matter if it goes all the way or not, right? Or maybe, we should allow deleting versions with DOI if we're wanting to delete the entire artifact anyway? I'm not too familiar with DOI, or the implications of deleting an artifact which is linked to one.

Reply:
I think the "delete all" button probably shouldn't even be there if any version has a DOI; deleting will be impossible necessarily.

Reply:
related to @Mark-Powers' comment, another arguably cleaner way to do this, since you're returning a bool from that function, is this:

```python
if not all(_delete_artifact_version(request, v) for v in artifact.versions):
    # handle not all of them deleting properly
```

That will attempt to delete any that it can.